services/events.js

```diff
@@ -0,0 +1,114 @@
+const path = require('path');
+const fs = require('fs');
+const { v4: uuidv4 } = require('uuid');
+const csv = require('fast-csv');
+// const lockfile = require('proper-lockfile');
+
+const headers = ['id', 'title', 'location', 'date', 'time', 'deleted'];
+const availableFieldsForFiltering = ['id', 'title', 'location', 'date', 'time'];
+const filterBy = (filterFields) => (row) => {
+  if (Boolean(+row.deleted)) {
+    return false;
+  }
+  return filterFields.every(([field, value]) => {
+    if (!availableFieldsForFiltering.includes(field)) {
+      return true;
+    }
+    return row[field] === value;
+  });
+};
+
+class EventsService {
+  filePath = path.resolve(__dirname, '..', 'storage', 'events.csv');
+  formatOptions = { headers, writeHeaders: true, includeEndRowDelimiter: true };
+
+  async findAll(filterFields = []) {
+    const events = await this.read();
```

Comment:
Potentially we could avoid loading all the events into the memory by filtering needed events while reading data from the file (using streams). The node process would use less memory.